internal/releasesjson/downloader.go

```diff
@@ -127,19 +114,38 @@ func (d *Downloader) DownloadAndUnpack(ctx context.Context, pv *ProductVersion,
 	pkgFilePath, err := filepath.Abs(pkgFile.Name())
 
 	d.Logger.Printf("copying %q (%d bytes) to %s", pb.Filename, expectedSize, pkgFile.Name())
-	// Unless the bytes were already downloaded above for checksum verification
-	// this may take a while depending on network connection as the io.Reader
-	// is expected to be http.Response.Body which streams the bytes
-	// on demand over the network.
-	bytesCopied, err := io.Copy(pkgFile, pkgReader)
-	if err != nil {
-		return pkgFilePath, err
+
+	var bytesCopied int64
+	if d.VerifyChecksum {
+		d.Logger.Printf("verifying checksum of %q", pb.Filename)
+		h := sha256.New()
+		r := io.TeeReader(resp.Body, pkgFile)
+
+		bytesCopied, err = io.Copy(h, r)
+		if err != nil {
+			return "", err
+		}
+
+		if !bytes.Equal(h.Sum(nil), verifiedChecksum) {
+			return pkgFilePath, fmt.Errorf(
+				"checksum mismatch (expected: %x, got: %x)",
+				verifiedChecksum, h.Sum(nil),
```

Comment:
```suggestion
		calculatedSum := h.Sum(nil)
		if !bytes.Equal(calculatedSum, verifiedChecksum) {
			return pkgFilePath, fmt.Errorf(
				"checksum mismatch (expected: %x, got: %x)",
				verifiedChecksum, calculatedSum,
```
Nitpick: Just to provide some extra clarity/readability here.